### PR TITLE
UX: expand threads list tap area on mobile

### PR DESCRIPTION
**Pull request by dbattersby**
This change adds a wrapper link around the thread list details on mobile to make the click area larger.

We also update child div elements to span to ensure valid html, since the link is an inline element and divs are block level.

### CODE REVIEW

File: plugins/chat/assets/javascripts/discourse/components/thread-title/index.gjs [hunk 1]
```diff
@@ -1,11 +1,14 @@
 import Component from "@glimmer/component";
 import { LinkTo } from "@ember/routing";
+import { inject as service } from "@ember/service";
```
Comment (jjaffeux) [inline plugins/chat/assets/javascripts/discourse/components/thread-title/index.gjs:3]:
```suggestion
import { service } from "@ember/service";
```

**Comment by jjaffeux** [comment]
@chapoi can you look at this please?

File: plugins/chat/assets/stylesheets/mobile/chat-user-threads.scss [hunk 2]
```diff
@@ -1,10 +1,7 @@
 .c-user-thread {
   display: grid;
-  grid-template-areas:
-    "avatar category timestamp"
-    "avatar title indicator"
-    "avatar excerpt excerpt";
-  grid-template-columns: auto 1fr auto;
+  grid-template-areas: "avatar details";
+  grid-template-columns: auto 1fr;
```
Comment (jjaffeux) [inline plugins/chat/assets/stylesheets/mobile/chat-user-threads.scss:4]: Do we even need these grid template areas here @chapoi ? That feels super overkill for what we are doing here and they are such a pain to read

Comment (dbattersby) [reply plugins/chat/assets/stylesheets/mobile/chat-user-threads.scss:4]: I had to make it a grid within a grid to get it to work with the newly wrapped `<a>` tag. I'm not completely sold on the grid layout approach (and why it's even needed) but I've adapted this to get it to work while retaining the grid template areas.

Comment (chapoi) [reply plugins/chat/assets/stylesheets/mobile/chat-user-threads.scss:4]: Usually I don't use grid unless I have to for either desktop/mobile conversion, or for alignments. I'll take a look but it's most likely not overkill.

Comment (chapoi) [reply plugins/chat/assets/stylesheets/mobile/chat-user-threads.scss:4]: Ah but I see now what you mean: there is a new grid lower in the hierarchy. Yes, than the previous grid is no longer needed and can be just flexed.

**Review by chapoi — APPROVED** [review]
LGTM

**Pull request merged**